### ISSUE TITLE
Need patch to use Drei with React Bootstrap

### ISSUE THREAD

**Issue by coobeeyon** [pmndrs/drei#1193]
- `three` version:0.146.0
- `@react-three/fiber` version:8.9.1
- `@react-three/drei` version:9.47.1
- `node` version:
- `npm` (or `yarn`) version:1.22.19

### Problem description:

Drei injects IntrinsicElements globally and this creates an incompatibility with React Bootstrap because it causes types to blow up into unions too large for the TS compiler to manage.  In some cases I see it adding things like Node<any, any> so I'm not entirely surprised.  This code sandbox shows a minimal repro: https://codesandbox.io/s/ts-compiler-error-repro-2xpkz0?file=/src/App.tsx

### Relevant code:
This code sandbox shows a minimal repro: https://codesandbox.io/s/ts-compiler-error-repro-2xpkz0?file=/src/App.tsx

```ts
import "@react-three/drei";
import { Button} from "react-bootstrap";
export default function App() {
  return <Button />;
}
```

### Suggested solution:
I think that if this injection is absolutely necessary the problem can be avoided by making sure that the new definitions of IntrinsicElements are much narrower in the types that they use.  This would be a real solution, not a workaround.

This can be worked around however with the attached patch which removes all global injections:
[@react-three+drei+9.47.1.zip](https://github.com/pmndrs/drei/files/10281503/%40react-three%2Bdrei%2B9.47.1.zip)

I think that if this injection is absolutely necessary the problem can be avoided by making sure that the new definitions of IntrinsicElements are much narrower in the types that they use.  This would be a real solution, not a workaround.

**Comment by drcmda** [comment]
I think you are just missing @types/three

**Comment by coobeeyon** [comment]
That doees not fix the problem either locally or in the sandbox: https://codesandbox.io/s/ts-compiler-error-repro-with-types-three-ze2heb?file=/src/App.tsx

**Comment by drcmda** [comment]
it is still not clear to me what the problem is, that is, if it's even ours or bootstraps. atm we just type our native elements, wich imo is not something i'd want to remove just like that if it can be avoided or remedied.

@CodyJasonBennett is this something we can fix? i saw you don't write into global intrinsic space no more, but in drei we still do, is that what's causing it?

**Comment by CodyJasonBennett** [comment]
We can try writing into `ThreeElements` in a major, but I think this would need a major refactor to narrow everything on top of pending work (related https://github.com/pmndrs/drei/issues/823).

**Comment by coobeeyon** [comment]
I had the feeling that some of the injections were not causing problems, but didn't do an exhaustive search.  This could mean that injecting narrower types could fix the problem until you have time to make the big switch.

**Comment by uhgtg** [comment]
Hey
Have you already solved the problem? Because I also have it.

**Comment by coobeeyon** [comment]
The patch I posted above is still working for us here as a workaround.

**Comment by uhgtg** [comment]
@coobeeyon can you please explain how I can apply the patch

**Comment by coobeeyon** [comment]
This is the method I use: https://dev.to/zhnedyalkow/the-easiest-way-to-patch-your-npm-package-4ece

**Comment by gucolin** [comment]
Still seeing this issue in 9.81.0. Applied the patch attached and still getting the same error for Buttons

**Comment by gucolin** [comment]
Still seeing this issue in 9.81.0. Applied the patch attached and still getting the same error for Buttons

**Comment by CodyJasonBennett** [comment]
Related #823.

@gucolin, did you refactor Drei components to use `ThreeElements` as per https://github.com/pmndrs/drei/issues/1193#issuecomment-1363320267 or simply remove them?

**Comment by xcibe95x** [comment]
Encountered the same issue today

**Comment by gucolin** [comment]
> Related #823.
> 
> @gucolin, did you refactor Drei components to use `ThreeElements` as per [#1193 (comment)](https://github.com/pmndrs/drei/issues/1193#issuecomment-1363320267) or simply remove them?

I just removed them.

**Comment by CodyJasonBennett** [comment]
This was fixed in R3F [v8.17.8](https://github.com/pmndrs/react-three-fiber/releases/tag/v8.17.8).
